EsqlClientYamlIT » test {p0=esql/60_enrich/Enrich on keyword}

### CI Link

https://gradle-enterprise.elastic.co/s/doqh4cuavfy7s/console-log?page=3#L2471

### Repro line

./gradlew ':x-pack:plugin:esql:qa:server:mixed-cluster:v8.12.3#yamlRestTest' -Dtests.class="org.elasticsearch.xpack.esql.qa.mixed.EsqlClientYamlIT" -Dtests.method="test {p0=esql/60_enrich/Enrich on keyword}" -Dtests.seed=19C4FD8ABFD05226 -Dtests.bwc=true -Dtests.locale=lv -Dtests.timezone=America/Indiana/Knox -Druntime.java=21

### Does it reproduce?

Yes

### Applicable branches

main

### Failure history

_No response_

### Failure excerpt

```java.lang.AssertionError: Failure at [esql/60_enrich:95]: expected [2xx] status code but api [esql.query] returned [500 Internal Server Error] [{"error":{"root_cause":[{"type":"unsupported_operation_exception","reason":"unsupported_operation_exception: can't load _source as part of enrich",```

--- a/x-pack/plugin/esql/qa/server/mixed-cluster/build.gradle
+++ b/x-pack/plugin/esql/qa/server/mixed-cluster/build.gradle
@@ -39,10 +39,6 @@ BuildParams.bwcVersions.withWireCompatible(supportedVersion) { bwcVersion, baseN
   def yamlRestTest = tasks.register("v${bwcVersion}#yamlRestTest", StandaloneRestIntegTestTask) {
     usesBwcDistribution(bwcVersion)
     systemProperty("tests.old_cluster_version", bwcVersion)
-    systemProperty("tests.rest.blacklist", [
-      // https://github.com/elastic/elasticsearch/issues/106507
-      "esql/60_enrich/Enrich on keyword"
-    ].join(','))
     testClassesDirs = sourceSets.yamlRestTest.output.classesDirs
     classpath = sourceSets.yamlRestTest.runtimeClasspath
   }

--- a/x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/esql/60_enrich.yml
+++ b/x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/esql/60_enrich.yml
@@ -94,7 +94,7 @@ teardown:
 "Enrich on keyword":
   - skip:
       version: " - 8.12.99"
-      reason: "AwaitsFix https://github.com/elastic/elasticsearch/issues/106507"
+      reason: "Enriching with text field in the enrich fields list involves reading text from source, not supported before 8.13.0"
       features: allowed_warnings_regex
 
   - do:

--- a/x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/esql/60_enrich.yml
+++ b/x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/esql/60_enrich.yml
@@ -92,6 +92,11 @@ teardown:
 
 ---
 "Enrich on keyword":
+  - skip:
+      version: " - 8.12.99"
+      reason: "AwaitsFix https://github.com/elastic/elasticsearch/issues/106507"
+      features: allowed_warnings_regex
+
   - do:
       allowed_warnings_regex:
         - "No limit defined, adding default limit of \\[.*\\]"

